blocks/library/embed/index.js

```diff
@@ -19,17 +19,21 @@ registerBlock( 'core/embed', {
 		caption: children( 'figcaption' )
 	},
 
-	edit( { attributes, isSelected, setAttributes } ) {
+	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, title, caption } = attributes;
 
 		return (
 			<figure>
-				<iframe src={ url } title={ title } />
-				{ caption || isSelected ? (
+				<div className="iframe-overlay">
+					<iframe src={ url } title={ title } />
```

Comment:
The height? I think it should work just the same as you have now, if you apply `position: relative;` to `.embed-wrapper` and `position: absolute; top: 0; right: 0; bottom: 0; left: 0;` to `.embed-overlay`.